Data/task.py

```diff
@@ -1,9 +1,30 @@
+from Data.future import Future
+import api_pb2
+
 class Task:
-    
+    # self.taskData is a list of future or/and bytes
     def __init__(self, taskId, taskDefintion, taskData):
         self.taskId = taskId
         self.taskDefintion = taskDefintion
-        self.taskData = taskData
+        if len(taskData)!=0 and not isinstance(taskData[0], api_pb2.TaskDataParam):
+            self.taskData = taskData
+        # Converts list of api_pb2.TaskDataParam to list of future and/or bytes
+        else:
+            self.taskData = []
+            for i in taskData:
+                if i.WhichOneof("dataParam") == "future":
```

Comment:
by this logic are we implying that taskData can have lets say 3 futures and 3 matrices

Reply:
Yes, taskData can have mix of futures and bytes(matrices), And the flow will still work.

Reply:
but when can taskData have both?

Reply:
It doesn't have to, its an additional support we provide.